x-pack/plugins/file_upload/public/components/index_settings.js

```diff
@@ -0,0 +1,169 @@
+        : [];
+      setIndexNames(indexNames);
+    });
+  }
+
+  if (!indexPatterns) {
+    getExistingIndexPatterns().then(
+      indexPatterns => setIndexPatterns(indexPatterns)
+    );
+  }
+
+  const indexDisabled = disabled || !indexTypes || !indexTypes.length;
+
+  return (
+    <Fragment>
+      <EuiSpacer size="m" />
+      <EuiFormRow
+        label={
+          <FormattedMessage
+            id="xpack.file_upload.indexNameLabel"
+            defaultMessage="Index type"
+          />
+        }
+      >
+        <EuiSelect
+          disabled={indexDisabled}
+          options={indexTypes.map(indexType =>({
+            text: indexType,
+            value: indexType,
+          }))}
+          onChange={({ target }) => setSelectedIndexType(target.value)}
+        />
+      </EuiFormRow>
+      <EuiSpacer size="m" />
+      {indexDisabled
+        ? null
+        : (
+          <EuiCallOut
+            title="Index name guidelines"
+            iconType="pin"
+          >
+            <div>
+              <ul>
+                <li>Must be a new index</li>
+                <li>Lowercase only</li>
+                <li>{`Cannot include \\, /, *, ?, ", <, >, |, \` \` \
+                  (space character), , (comma), #`
+                }
+                </li>
+                <li>{`Cannot start with -, _, +`}</li>
+                <li>{`Cannot be . or ..`}</li>
+                <li>{
+                  `Cannot be longer than 255 bytes (note it is bytes, \
+                  so multi-byte characters will count towards the 255 \
+                  limit faster)`
+                }
+                </li>
+              </ul>
+            </div>
+          </EuiCallOut>
+        )}
+      <EuiSpacer size="s" />
+      <EuiFormRow
+        label={
+          <FormattedMessage
+            id="xpack.file_upload.indexNameLabel"
+            defaultMessage="Index name"
+          />
+        }
+        isInvalid={indexNameError !== ''}
+        error={[indexNameError]}
+      >
+        <EuiFieldText
+          disabled={indexDisabled}
+          placeholder={intl.formatMessage({
+            id: 'xpack.file_upload.indexNamePlaceholder',
+            defaultMessage: 'Enter Index Name'
+          })}
+          value={indexName}
+          onChange={onIndexChange(setIndexName, setIndexNameError, indexNames)}
+          isInvalid={indexNameError !== ''}
+          aria-label={intl.formatMessage({
+            id: 'xpack.file_upload.indexNameAriaLabel',
+            defaultMessage: 'Index name, required field'
+          })}
+        />
+      </EuiFormRow>
+
+      <EuiSpacer size="s" />
+
+    </Fragment>
+  );
+});
+
+function onIndexChange(setIndex, setIndexNameError, indexNames) {
+  return ({ target }) => {
+    const name = target.value;
+    const errorMessage = isIndexNameValid(name, indexNames);
+    setIndexNameError(errorMessage);
+    setIndex(name);
+  };
+}
+
+function isIndexNameValid(name, indexNames) {
```

Comment:
this should also check if the name does not conflict with an existing index-pattern